### ISSUE TITLE
VDC Helm chart - update connector versioning logic

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#2696]
We have a VDC helm chart that deploys egeria, atlas, ranger, gaian etc.
Whilst we intend to focus on this more in the 1.8 timeframe according to the current release plan, there are people using it currently, so we continue to ensure it keeps working as-is.

When a release of egeria is made all references to a release version like '1.5-SNAPSHOT' are changed to '1.5'. This is a full text change so covers poms, but also readmes & source files.

In the helm chart definition for vdc currently we have a few areas to consider
```
egeria:
  version: 1.5
```
This is the version to be used for the docker containers (egeria, configure, jupyter) - and is updated correctly.

There is an issue open to create this version of container during the release cycle automatically
 ( #2583 )

```
connectors:
  repo: snapshot
  version: 1.5
```
This requires consideration. The global change has updated the version tag to 1.5, but this will not immediately be available -- until @cmgrote has had a change to test

Additionally the repo once releases should be 'release'

The easiest option IMO would be to use some scripting in our templates to automatically determine whether to use 'snapshot' or 'release' as the *jfrog* repo - this could be based on whether the version includes '-SNAPSHOT' (and dd a note to the config file/readme to warn the connector may not be immediately available ). This is a small change, and probably the simplest, albeit at the exposure of a few days where the VDC chart doesn't work

There are other options that might include changing in which project the helm charts are placed, their build process, what containers we use, repositories for artifacts, but this is probably overkill for now especially as the vdc chart is going to be revisited in a few releases
```
# Default values used for images
# 'tag' is only used for images other than connectors/egeria where the version above is used
imageDefaults:
  tag: latest
```
This remains appropriate - for atlas, ranger etc

My *short term* proposal for 1.5 is to change repo from 'snapshot' to 'release' so that it works 'soon' (I've published the 1.5 docker images early - chicken & egg challenge). This needs to be done imminently (I will do this)

It would be good to determine the right way moving forward for 1.6/7

cc: @cmgrote

**Comment by cmgrote** [comment]
Part of the reason I've left the `repo` section separate rather than auto-determining it is that during final testing it is useful to use Artifactory's staged artifacts (prior to their release), in which case the `repo` is `staging` while the release identifier is just `1.x` (ie. no `-SNAPSHOT`, but also not in the `release` repo, yet...)

I guess we could have a complex statement that first looks for the `release`, then the `staging`, then falls back to the `snapshot` -- this just makes it difficult to identify which artifact is actually used in the chart someone is running, in case we need to troubleshoot something...

**Comment by planetf1** [comment]
We release egeria, and once done, those default values will not change - at least as we're currently structured. We'd also ideally like the chart to work 'out of the box' without changes

What we set for development/testing is less of a concern - we do need more flexibility, though having less changes is easier (and means we testing closer to what the end user gets)

So how do we handle that

It looks as if setting the 'future' value is problematic due to the inevitable lag

Other options:
 * We could always set to the previous version (though this will be an extra thing to update prior to release)
 * We could have a more complex script to pull the 'right' connector. The values.yaml could have a value of 'auto' but it could be overwritten with a specific value (similar to what you refer to)
 * We REQUIRE the end user to override the connector version if using IGC or atlas
 * We always just get the latest released version 

We need both an immediate decision for 1.5, as well as any more complex fixes for 1.6 (ideally)

**Comment by cmgrote** [comment]
Per Slack discussion, decided:

- for 1.5 release, we'll just stick to using `1.5-SNAPSHOT` for the connectors (that way they'll at least still work)
- for 1.6 release onwards, I'll attempt to implement the precedence-based retrieval as follows: prefer firstly the released version of a connector, if not available then fallback to the staged version of a connector, and if that is not available then fallback all the way to the snapshot version of the connector
- we will be able to see which version of the connector was actually downloaded by looking at either the configmap or the environment variables that are set (from the configmap) within the container itself -- the `CONNECTOR_URL` will show the precise location of the connector that was retrieved (whether release, staging or snapshot)

**Comment by planetf1** [comment]
Agreed - I've merged in the 1.5 update and tested.
